pkg/sippyserver/pr_commenting_processor.go

```diff
@@ -555,6 +543,71 @@ func (aw *AnalysisWorker) processRiskAnalysisComment(prPendingComment models.Pul
 	log.Debugf("Comment added to pendingComments: %s/%s/%s", pendingComment.org, pendingComment.repo, pendingComment.sha)
 }
 
+func buildComment(sortedAnalysis RiskAnalysisEntryList, sha string) string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("Job Failure Risk Analysis for sha: %s\n\n| Job Name | Failure Risk |\n|:---|:---|\n", sha))
+
+	// don't want the comment to be too large so if we have a high number of jobs to analyze
+	// reduce the max tests / reasons we show
+	maxSubRows := 3
+	if len(sortedAnalysis) > 10 {
+		maxSubRows = 1
+	}
+
+	for a, value := range sortedAnalysis {
+		tableKey := value.Key
+
+		// top 20 should be more than enough
+		if a > 19 {
+			sb.WriteString(fmt.Sprintf("\nShowing %d of %d jobs analysis", a, len(sortedAnalysis)))
+			break
+		}
+
+		if value.Value.URL != "" {
+			tableKey = fmt.Sprintf("[%s](%s)", value.Key, value.Value.URL)
+		}
+
+		var riskSb strings.Builder
+		riskSb.WriteString(fmt.Sprintf("**%s**", value.Value.RiskLevel.Name))
+		for i, t := range value.Value.TestRiskAnalysis {
+			if i > maxSubRows {
+				riskSb.WriteString(fmt.Sprintf("<br>---<br>Showing %d of %d test results", i, len(value.Value.TestRiskAnalysis)))
+				break
+			}
+			if i > 0 {
+				riskSb.WriteString("<br>---")
+			}
+			riskSb.WriteString(fmt.Sprintf("<br>*%s*", t.Name))
+			for j, r := range t.Risk.Reasons {
+				if j > maxSubRows {
+					riskSb.WriteString(fmt.Sprintf("<br>Showing %d of %d test risk reasons", j, len(t.Risk.Reasons)))
+					break
+				}
+				riskSb.WriteString(fmt.Sprintf("<br>%s", r))
+			}
+
+			// Do we have open bugs?  Stack them vertically to preserve real estate
+			for k, b := range t.OpenBugs {
+
+				// Currently we don't limit the number of open bugs we show
+				if k == 0 {
+					if len(t.Risk.Reasons) > 0 {
+						riskSb.WriteString("<br><br>")
+					}
+
+					riskSb.WriteString("Open Bugs")
+				}
+				riskSb.WriteString(fmt.Sprintf("<br>%s", b.URL))
```

Comment:
Sorry one more req, can we have bug title as link body.

Reply:
np